torch/nn/parallel/distributed.py

```diff
@@ -727,7 +745,8 @@ def __setstate__(self, state):
         # Builds reducer
         self._ddp_init_helper(parameters, expect_sparse_gradient, param_to_name_mapping)
         if self.static_graph:
-            self._set_static_graph()
+            self.reducer._set_static_graph()
```

Comment:
Why can't we just call `self._set_static_graph()` here?

Reply:
since I let _set_static_graph() early return if self.static_graph=True, to make it work even users set static_graph=True and accidentally call _set_static_graph() again.  

because of this change, calling self. _set_static_graph() will early return, not really doing anything here